src/EventLogExpert.Eventing/EventResolvers/EventProviderDatabaseEventResolver.cs

```diff
@@ -166,19 +161,17 @@ public DisplayEventModel Resolve(EventRecord eventRecord, string OwningLogName)
 
             try
             {
-                foreach (var dbContext in dbContexts)
+                foreach (var dbContext in _dbContexts)
                 {
-                    var providerDetails = dbContext.ProviderDetails.FirstOrDefault(p => p.ProviderName.ToLower() == eventRecord.ProviderName.ToLower());
+                    var providerDetails = dbContext.ProviderDetails?.FirstOrDefault(p => p.ProviderName.Equals(eventRecord.ProviderName, StringComparison.CurrentCultureIgnoreCase));
                     if (providerDetails != null)
                     {
-                        lastResult = ResolveFromProviderDetails(eventRecord, eventProperties, providerDetails, OwningLogName);
-
-                        if (lastResult?.Description != null)
-                        {
-                            _tracer($"Resolved {eventRecord.ProviderName} provider from database {dbContext.Name}.", LogLevel.Information);
-                            _providerDetails.TryAdd(eventRecord.ProviderName, providerDetails);
-                            return lastResult;
-                        }
+                        lastResult = ResolveFromProviderDetails(eventRecord, eventProperties, providerDetails, owningLogName);
+
+                        tracer($"Resolved {eventRecord.ProviderName} provider from database {dbContext.Name}.", LogLevel.Information);
+                        _providerDetails.TryAdd(eventRecord.ProviderName, providerDetails);
+                        
+                        return lastResult;
```

Comment:
We should only accept this as a result if lastResult is not null and Description is not null. Otherwise, we need to continue with the rest of the logic. But those checks are now gone. This will break event resolution for some scenarios.